tests/unittests/config/test_cc_ca_certs.py

```diff
@@ -62,30 +62,35 @@ def _fetch_distro(self, kind):
         paths = helpers.Paths({})
         return cls(kind, {}, paths)
 
-    def _mock_init(self):
+    def _mock_init(self, distro_name):
         self.mocks = ExitStack()
         self.addCleanup(self.mocks.close)
 
         # Mock out the functions that actually modify the system
         self.mock_add = self.mocks.enter_context(
             mock.patch.object(cc_ca_certs, "add_ca_certs")
         )
+        self.mock_remove = self.mocks.enter_context(
+            mock.patch.object(cc_ca_certs, "disable_default_ca_certs")
+        )
         self.mock_update = self.mocks.enter_context(
             mock.patch.object(cc_ca_certs, "update_ca_certs")
         )
-        self.mock_remove = self.mocks.enter_context(
-            mock.patch.object(cc_ca_certs, "remove_default_ca_certs")
-        )
+        if distro_name == "freebsd":
+            mock.patch(
+                "cloudinit.distros.networking.subp.subp",
+                return_value=("", None),
+            )
 
     def test_no_trusted_list(self):
         """
         Test that no certificates are written if the 'trusted' key is not
         present.
         """
-        config = {"ca-certs": {}}
+        config = {"ca_certs": {}}
 
         for distro_name in cc_ca_certs.distros:
-            self._mock_init()
+            self._mock_init(distro_name)
             cloud = get_cloud(distro_name)
```

Comment:
the instanciation happens here, so this is where the mock needs to go

Reply:
That's right after where _mock_init is called (in several testcases) so it made sense to put the mock in a single place in _mock_init. I had previously tried putting it here but it also had no effect.

Reply:
that's interesting! that should not be happening!